<a href="https://colab.research.google.com/github/Ridhof177/Tugas-Mata-Kuliah-Teknik-Pengambilan-Sampel-dan-Data-Wrangling./blob/main/RIDHO_FATILLA_WIBAWA_241061010_SAINS_DATA_24_TUGAS_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

###DATA BELANJA PELANGGAN

1. DOWNLOAD: CARI DATASET BEBAS DI KAGGLE
- https://www.kaggle.com/datasets/datascientistanna/customers-dataset

2. BACA DATASET TERSEBUT DI GOOGLE COLAB ATAU JUPYTER LAB

In [8]:
import pandas as pd
import json

sd = pd.read_csv("/content/Customers.csv")

3. TAMPILKAN DATASET MENGGUNAKAN PERINTAH HEAD(), INFO(), DAN SHAPE.

In [20]:
sd.head()
#penjelasan di cell selanjutnya

,CustomerID,Gender,Age,Annual Income ($),Spending Score (1-100),Profession,Work Experience,Family Size
0,1,Male,19,15000,39,Healthcare,1,4
1,2,Male,21,35000,81,Engineer,3,3
2,3,Female,20,86000,6,Engineer,1,1
3,4,Female,23,59000,77,Lawyer,0,2
4,5,Female,31,38000,40,Entertainment,2,6


- Perintah bisa sd.head(N) dengan N adalah berapa baris yang akan ditampilkan.
- Perintah head() digunakan untuk menampilkan beberapa baris pertama dari dataset pada objek DataFrame (sd).
- Secara default → menampilkan 5 baris teratas

In [19]:
sd.tail()
#penjelasan di cell selanjutnya

,CustomerID,Gender,Age,Annual Income ($),Spending Score (1-100),Profession,Work Experience,Family Size
1995,1996,Female,71,184387,40,Artist,8,7
1996,1997,Female,91,73158,32,Doctor,7,7
1997,1998,Male,87,90961,14,Healthcare,9,2
1998,1999,Male,77,182109,4,Executive,7,2
1999,2000,Male,90,110610,52,Entertainment,5,2


- Perintah bisa sd.tail(N) dengan N adalah berapa baris terakhir yang akan ditampilkan.
- Perintah tail() digunakan untuk menampilkan beberapa baris terakhir dari dataset pada objek DataFrame (sd).
- Secara default → menampilkan 5 baris paling bawah dari dataset.

In [23]:
sd.info()
#penjelasan di cell selanjutnya

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 8 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   CustomerID              2000 non-null   int64 
 1   Gender                  2000 non-null   object
 2   Age                     2000 non-null   int64 
 3   Annual Income ($)       2000 non-null   int64 
 4   Spending Score (1-100)  2000 non-null   int64 
 5   Profession              1965 non-null   object
 6   Work Experience         2000 non-null   int64 
 7   Family Size             2000 non-null   int64 
dtypes: int64(6), object(2)
memory usage: 125.1+ KB


- RangeIndex: 2000 entries: Menunjukkan total baris data sebanyak 2.000 baris (indeks 0 sampai 1999).
- Data columns (total 8 columns): Dataset memiliki 8 kolom variabel.
- Non-Null Count: Menunjukkan jumlah data yang terisi pada setiap kolom; kolom Profession memiliki 1965 data sehingga terdapat beberapa missing values.
- Dtypes: Terdapat tipe data object (Gender, Profession) dan int64 (CustomerID, Age, Annual Income, Spending Score, Work Experience, Family Size).
- Memory usage: Menunjukkan penggunaan memori dataset sebesar sekitar 125.1 KB.

In [22]:
sd.shape
#penjelasan di cell selanjutnya

(2000, 8)

- sd.shape: Digunakan untuk mengetahui ukuran atau dimensi dataset pada DataFrame sd.
- Output menunjukkan jumlah baris dan jumlah kolom dalam bentuk (baris, kolom).
- Berdasarkan data, sd memiliki 2000 baris dan 8 kolom sehingga hasilnya (2000, 8).

4. jelaskan struktur dataset tersebut (jumlah baris, kolom, dan tipe data)

Dataset customers.csv memiliki struktur data yang terdiri dari 2000 baris (entries) dan 8 kolom (variabel). Setiap baris merepresentasikan satu data pelanggan, sedangkan setiap kolom berisi atribut informasi pelanggan seperti CustomerID, Gender, Age, Annual Income ($), Spending Score (1–100), Profession, Work Experience, dan Family Size. Berdasarkan tipe datanya, dataset ini memiliki dua jenis tipe data yaitu int64 untuk data numerik (CustomerID, Age, Annual Income, Spending Score, Work Experience, dan Family Size) serta object untuk data kategorik atau teks (Gender dan Profession). Struktur ini menunjukkan bahwa dataset berisi kombinasi data numerik dan kategorikal yang dapat digunakan untuk analisis data pelanggan.

5. buatkan struktur data dengan format json, secara manual dan konversi menjadi dataset dalam format csv (dataset hasil konversi ke csv di upload di github)

In [32]:
# 1. Data JSON
data = [
    {"ID": 1, "Nama": "Andi", "Umur": 30, "Kota": "Makassar"},
    {"ID": 2, "Nama": "Siti", "Umur": 32, "Kota": "Jakarta"},
    {"ID": 3, "Nama": "Budi", "Umur": 31, "Kota": "pare-pare"}
]

# 2. Simpan sebagai file JSON
with open("data.json", "w") as file:
    json.dump(data, file, indent=4)

# 3. Konversi JSON ke DataFrame
df = pd.DataFrame(data)

# 4. Simpan menjadi CSV
df.to_csv("data_peserta.csv", index=False)

print("Konversi JSON ke CSV berhasil!")

Konversi JSON ke CSV berhasil!
